example/src/App.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useRef } from 'react'
+
+import {
+  PtsCanvas,
+  HandleStartFn,
+  HandleAnimateFn,
+  HandleActionFn
+} from 'react-pts-canvas'
+import { Pt, Rectangle, Group, Geom, Create, Num, Shaping } from 'pts'
+import './App.css'
+
+const ExampleComponent: React.FC = () => {
+  const gridRef = useRef<Group[] | null>(null)
+  const handleStart: HandleStartFn = (_bound, space) => {
+    if (!space) return
+    gridRef.current = Create.gridCells(
+      space.innerBound,
+      Math.floor(space.width / 50),
+      Math.floor(space.height / 50)
+    )
+  }
+
+  const innerRect = (rectPts: Pt[], t: number, d: number) => {
+    const lines = rectPts.map((r, i) =>
+      i >= 3 ? new Group(r, rectPts[0]) : new Group(r, rectPts[i + 1])
+    )
+    return lines.map(ln =>
+      Geom.interpolate(ln[d > 0 ? 0 : 1], ln[d > 0 ? 1 : 0], t)
+    )
+  }
+
+  const handleAnimate: HandleAnimateFn = (space, form, time) => {
+    if (!form || !space || !gridRef.current || time === undefined) return
+    const t = Shaping.cubicInOut((time % 3000) / 3000, 1)
+    const colors = ['#fff', '#62f', '#f03']
+    const d = space.pointer.x > space.center.x ? 1 : -1
+    gridRef.current.forEach((g, i) => {
+      const r = innerRect(
+        Rectangle.corners(g),
+        Num.boundValue(t + i * d * 0.004, 0, 1),
+        d
+      )
+      form.fillOnly(colors[i % colors.length]).polygon(r)
+    })
+  }
+
+  return (
+    <PtsCanvas
+      background="#0c9"
+      name="pts-tester"
+      style={{ opacity: 0.95 }}
+      onStart={handleStart}
+      onAnimate={handleAnimate}
+    />
+  )
+}
+
+const ExampleComponent2: React.FC = () => {
+  // Use a useRef instead of useState hook here as the canvas
+  // Doesn't re-render based on the radius, it only affects the handleAnimate
+  const radiusRef = useRef(50)
```

Comment:
This ref approach for mutable variables is probably correct, but it feels heavy... is there an alternative approach where we can use `let radius = 50`? If there is, it would be good to have an example here too.